### PR TITLE
Allow error in create-dmg

### LINKED ISSUE #984: create-dmg non-zero exit breaks Mac builds
The problem is in `create-dmg`'s cleanup step. https://github.com/create-dmg/create-dmg/issues/133

It seems the DMG is created just fine. So in the interest of passing builds, we can probably force our workflow to continue past that failure.

### PR DESCRIPTION
**Pull request by davemfish**
There's consistently an error from `create-dmg` when it tries to unmount a volume. Until that's fixed upstream, we can make sure our workflows continue past this step.

Fixes #984 

## Checklist
- [ ] Updated HISTORY.rst (if these changes are user-facing)
- [ ] Updated the user's guide (if needed)
- [ ] Tested the affected models' UIs (if relevant)

### CODE REVIEW

**Comment by emlys** [comment]
@davemfish this is great. I wonder if we might also add a shorter timeout to the dmg building step? Looks like it took over 30 minutes, I assume most of that is waiting for the detach which is going to fail anyway.

**Comment by davemfish** [comment]
> @davemfish this is great. I wonder if we might also add a shorter timeout to the dmg building step? Looks like it took over 30 minutes, I assume most of that is waiting for the detach which is going to fail anyway.

Great idea! Looking back at past builds, I saw some taking ~10 minutes. So I set the timeout for `make binaries` at 15.

File: .github/workflows/build-and-test.yml [hunk 1]
```diff
@@ -376,6 +376,10 @@ jobs:
         run: make install
 
       - name: Build binaries
+        # Due to a create-dmg issue, sometimes we wait a very long time
+        # for a disk to unmount, and it fails to do so. Instead of waiting
+        # for an inevitable failure, set timeout-minutes.
+        timeout-minutes: 15
```
Comment (emlys) [inline .github/workflows/build-and-test.yml:382]:
```suggestion
        timeout-minutes: 15
        continue-on-error: true
```

**Review by emlys — CHANGES_REQUESTED** [review]
Ah I guess the timeout causes it to fail at the workflow level, maybe `continue-on-error` will fix it?

**Comment by davemfish** [comment]
> Ah I guess the timeout causes it to fail at the workflow level, maybe `continue-on-error` will fix it?

Ah thanks for noticing that. I added `continue-on-error` and removed the Makefile changes so we're only addressing this in one place. Presumably we'll always timeout now before `create-dmg` raises the error anyway.

**Comment by davemfish** [comment]
Woah 30 mins to restore the node modules cache for Mac, 30 seconds for Windows!

**Review by emlys — APPROVED** [review]

**Pull request merged**